### PR TITLE
Added flags and tests for createDecoder(). #24

### PR DESCRIPTION
**Pull request by emilelm96**
Now there are flags and coverage checking for createDecoder().

### CODE REVIEW

**Review by Jonagoo — APPROVED** [review]

File: src/main/java/com/jsoniter/extra/GsonCompatibilityMode.java [hunk 1]
```diff
@@ -342,100 +345,127 @@ public Object decode(JsonIterator iter) throws IOException {
                         String input = iter.readString();
```
Comment (apeinot) [inline src/main/java/com/jsoniter/extra/GsonCompatibilityMode.java:368]: I think you should add flag into the try

**Review by apeinot — CHANGES_REQUESTED** [review]
I think you should add two more flags:
- one into the try
- one before the last return of the function

**Comment by francislaus** [comment]
> I think you should add two more flags:
> 
>     * one into the try
> 
>     * one before the last return of the function

I think there should be no flag in the try scope as it is not a branch itself. It gets executed in all cases

**Comment by emilelm96** [comment]
Could we get some confirmation on this, @Samuelhertzberg or @francislaus?

**Review by francislaus — APPROVED** [review]
After further review, I think that it is good as it currently is

**Comment by Samuelhertzberg** [comment]
What about `iter.readBoolean() ? "true" : "false";` ?

**Review by apeinot — APPROVED** [review]

**Review by Samuelhertzberg — APPROVED** [review]

**Pull request merged**